codex-rs/tui/src/resume_picker.rs

```diff
@@ -0,0 +1,453 @@
+use std::path::Path;
+use std::path::PathBuf;
+
+use chrono::DateTime;
+use chrono::TimeZone;
+use chrono::Utc;
+use codex_core::ConversationItem;
+use codex_core::ConversationsPage;
+use codex_core::Cursor;
+use codex_core::RolloutRecorder;
+use codex_core::protocol::InputMessageKind;
+use color_eyre::eyre::Result;
+use crossterm::event::KeyCode;
+use crossterm::event::KeyEvent;
+use crossterm::event::KeyEventKind;
+use ratatui::layout::Constraint;
+use ratatui::layout::Layout;
+use ratatui::layout::Rect;
+use ratatui::style::Stylize as _;
+use ratatui::text::Line;
+
+use crate::text_formatting::truncate_text;
+use crate::tui::FrameRequester;
+use crate::tui::Tui;
+use crate::tui::TuiEvent;
+
+#[derive(Debug, Clone)]
+pub enum ResumeSelection {
+    StartFresh,
+    Resume(PathBuf),
+    Exit,
+}
+
+/// Interactive session picker that lists recorded rollout files with simple
+/// search and pagination. Shows the first user input as the preview, relative
+/// time (e.g., "5 seconds ago"), and the absolute path.
+pub async fn run_resume_picker(tui: &mut Tui, codex_home: &Path) -> Result<ResumeSelection> {
+    let _ = tui.enter_alt_screen();
```

Comment:
I feel like this should be restructured as:

```
let _ = tui.enter_alt_screen();
{
  // block that cannot early exit
}
let _ = tui.leave_alt_screen();
```

The early return and having to remember to call `let _ = tui.leave_alt_screen();` before it feels like it's easy to make a mistake.

Alternatively, having some struct that calls `tui.leave_alt_screen()` on `Drop` would be more idiomatic.